jerry-core/CMakeLists.txt

```diff
@@ -145,10 +145,7 @@ endif()
 # Valgrind
 set(INCLUDE_THIRD_PARTY_VALGRIND "${CMAKE_SOURCE_DIR}/third-party/valgrind")
 
-# build mode specific compile/link flags
-if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
-  set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_NDEBUG)
-endif()
+set (DEFINES_JERRY "$<$<NOT:$<CONFIG:Debug>>:JERRY_NDEBUG>")
```

Comment:
Why is this change needed? The two expressions seem to be (roughly) equivalent and the change seems to be unrelated to Windows support. I'd not support the rewrite if not justified. (Moreover it's actually not equivalent, `set(DEFINES_JERRY ${DEFINES_JERRY} "$<$<NOT:$<CONFIG:Debug>>:JERRY_NDEBUG>")` would be.)

Reply:
I've found the same SO post last night. CMake's multi-configuration generators are tricky to set config at build time, it's quite unexpected. I'm still not completely getting it how the evaluation of generator expressions can be postponed until build time, but let's accept that this is the suggested approach.

This change still deserves some comments:
- style: no space between set and (
- style: I'm not sure the quotes are necessary around the generator expression
- logic: as mentioned above, the rewrite is not functionally equivalent; an equivalent rewrite is `set(DEFINES_JERRY ${DEFINES_JERRY} $<$<NOT:$<CONFIG:Debug>>:JERRY_NDEBUG>)`